yt/geometry/coordinates/cartesian_coordinates.py

```diff
@@ -22,12 +22,48 @@
     cartesian_to_cylindrical, \
     cylindrical_to_cartesian
 from yt.funcs import mylog
+from yt.units.yt_array import uvstack, YTArray
 from yt.utilities.lib.pixelization_routines import \
     pixelize_element_mesh, pixelize_off_axis_cartesian, \
-    pixelize_cartesian, pixelize_cartesian_nodal
+    pixelize_cartesian, pixelize_cartesian_nodal, \
+    pixelize_element_mesh_line
 from yt.data_objects.unstructured_mesh import SemiStructuredMesh
 from yt.utilities.nodal_data_utils import get_nodal_data
 
+def _sample_ray(ray, npoints, field):
+    """
+    Private function that uses a ray object for calculating the field values
+    that will be the y-axis values in a LinePlot object.
+
+    Parameters
+    ----------
+    ray : YTOrthoRay, YTRay, or LightRay
+        Ray object from which to sample field values
+    npoints : int
+        The number of points to sample
+    field : str or field tuple
+        The name of the field to sample
+    """
+    start_point = ray.start_point
+    end_point = ray.end_point
+    sample_dr = (end_point - start_point)/(npoints-1)
+    sample_points = [np.arange(npoints)*sample_dr[i] for i in range(3)]
```

Comment:
I'm also a bit confused as to why `sample_dr = (end_point - start_point)/(npoints-1)` and not `sample_dr = (end_point - start_point)/npoints`.